feature_engine/creation/binary_combination_feature_reference.py

```diff
@@ -0,0 +1,250 @@
+                    "remove those before using this transformer."
+            )
+
+        result = []
+        result_name = []
+        for operation in self.binary_operations:
+            for combine in self.variables_to_combine:
+                for reference in self.reference_variables:
+                    if operation == "Sub":
+                            result.append(X[combine].values - X[reference].values)  
+                            result_name.append(str(combine)+"_"+operation+"_"+str(reference))
+                    elif operation == "Div":
+                            result.append(X[combine].values / X[reference].values)
+                            result_name.append(str(combine)+"_"+operation+"_"+str(reference))
+                    else: 
+                        result.append("At least one of the entered binary_operations is not supported. ")
+                        result_name.append(str(combine)+"_"+operation+"_"+str(reference))
+
+        #replace created variable names with user ones.
+        if len(self.new_variables_names) <= 0:
+            self.new_variables_names = result_name
+            
+        # to convert lists to dictionary 
+        combination_dict_ = dict(zip(self.new_variables_names, result)) 
+
+        self.input_shape_ = X.shape
+
+        return self
+
+    def transform(self, X: pd.DataFrame) -> pd.DataFrame:
+        """
+        Transforms source dataset.
+
+        Adds a column for each operation with the calculation based on the variables
+        and operations indicated when setting up the transformer.
+
+        Args:
+            X: pandas dataframe of shape = [n_samples, n_features]
+            The data to transform.
+
+        Returns:
+            Pandas dataframe, shape = [n_samples, n_features + n_operations]
+            The dataframe with the operations results added as columns.
+        """
+
+        # Check method fit has been called
+        check_is_fitted(self)
+
+        # check that input is a dataframe
+        X = _is_dataframe(X)
+
+        # check if dataset contains na
+        _check_contains_na(X, self.reference_variables)
+        _check_contains_na(X, self.variables_to_combine)
+
+        # cannot divide by 0, as will result in error
+        if 'sub' in self.binary_operations:
+            if X[self.reference_variables].isin([0]).any().any():
+                raise ValueError(
+                    "Some of the refence variables contain 0 values. Check and "
+                    "remove those before using this transformer."
+            )
+
+        # Check if input data contains same number of columns as dataframe used to fit.
+        _check_input_matches_training_df(X, self.input_shape_[1])
+ 
+        # Add new features and values into de data frame.
+        for new_variable_name, result in self.combination_dict_.items():
+            X[new_variable_name] = X[new_variable_name].agg(result, axis=1)
+
```

Comment:
the transform method should perform the transformation.

The fit method, for this transformer, doesn't do much, just performs a few checks